### ISSUE TITLE
RT baseline check is failing for coastal_scituateharbor_atm2fvc

### ISSUE THREAD

**Issue by uturuncoglu** [oceanmodeling/FVCOM#2]
@janahaddad @pvelissariou1 @saeed-moghimi-noaa I created baselines for supported test on Hercules and run the tests against baseline. The `coastal_scituateharbor_atm2fvc` test is failing as following,

```
==> logs/log_hercules/rt_002_coastal_scituateharbor_atm2fvc_intel.log <==

baseline dir = /work2/noaa/nems/tufuk/RT/NEMSfv3gfs/develop-20240126/coastal_scituateharbor_atm2fvc_intel
working dir  = /work2/noaa/stmp/tufuk/stmp/tufuk/FV3_RT/rt_732961/coastal_scituateharbor_atm2fvc_intel
Checking test 002 coastal_scituateharbor_atm2fvc_intel results ....
 Comparing OUTPUT/sci_0001.nc ............ALT CHECK......NOT OK
 Comparing OUTPUT/sci_restart_0001.nc ............ALT CHECK......NOT OK


Test 002 coastal_scituateharbor_atm2fvc_intel FAIL
```

It seems that we have reproducibility issue in here that needs to be solved. I don't think this configuration is ever tested agains the baseline before.

**Comment by uturuncoglu** [comment]
I also checked this one with NCAR's `cprnc` tool and lots of fields differ. I could not check the mediator history file since they are not included into the baseline but I'll change the test and add mediator history field to see them.

**Comment by uturuncoglu** [comment]
I check it again by including the mediator history file and it seems that we don't have issue with mediator history files. So, this shows that the data going to FVCOM is same but the answer is changing in the model side. I need to check the model cap and internal to see what is going on there but at least we could put mediator and data atmosphere out of picture.

```
baseline dir = /work2/noaa/nems/tufuk/RT/NEMSfv3gfs/develop-20240126/coastal_scituateharbor_atm2fvc_intel
working dir  = /work2/noaa/stmp/tufuk/stmp/tufuk/FV3_RT/rt_1112332/coastal_scituateharbor_atm2fvc_intel
Checking test 001 coastal_scituateharbor_atm2fvc_intel results ....
 Comparing OUTPUT/sci_0001.nc ............ALT CHECK......NOT OK
 Comparing OUTPUT/sci_restart_0001.nc ............ALT CHECK......NOT OK
 Comparing sci.cpl.hi.2022-06-22-00000.nc .........OK
 Comparing sci.cpl.hi.2022-06-23-00000.nc .........OK
 Comparing sci.cpl.hi.2022-06-24-00000.nc .........OK
```

**Comment by uturuncoglu** [comment]
This also failing on Orion agains the baseline,

```
baseline dir = /work/noaa/nems/tufuk/RT/NEMSfv3gfs/develop-20240126/coastal_scituateharbor_atm2fvc_intel
working dir  = /work/noaa/stmp/tufuk/stmp/tufuk/FV3_RT/rt_100793/coastal_scituateharbor_atm2fvc_intel
Checking test 002 coastal_scituateharbor_atm2fvc_intel results ....
 Comparing OUTPUT/sci_0001.nc ............ALT CHECK......NOT OK
 Comparing OUTPUT/sci_restart_0001.nc ............ALT CHECK......NOT OK
 Comparing sci.cpl.hi.2022-06-22-00000.nc .........OK
 Comparing sci.cpl.hi.2022-06-23-00000.nc .........OK
 Comparing sci.cpl.hi.2022-06-24-00000.nc .........OK

Binary file /work/noaa/stmp/tufuk/stmp/tufuk/FV3_RT/rt_100793/coastal_scituateharbor_atm2fvc_intel/out matches
Binary file /work/noaa/stmp/tufuk/stmp/tufuk/FV3_RT/rt_100793/coastal_scituateharbor_atm2fvc_intel/out matches

Test 002 coastal_scituateharbor_atm2fvc_intel FAIL
```

**Comment by janahaddad** [comment]
@uturuncoglu adding these FVCOM issues to ufs-coastal-project backlog for now...

**Comment by uturuncoglu** [comment]
@janahaddad Thanks. It sounds fine.

**Comment by uturuncoglu** [comment]
I check the first and second time step output of the model. The initial timesetp all the data is identical with the baseline but in the second time step `ua` and `va` fields have some difference. 

```
 ua   (nele,time)  t_index =      1     1
          6    11153  (  1794,     1) (  7958,     1) ( 11033,     1) ( 11033,     1)
               11153   5.447403192520142E-01  -4.624237418174744E-01 1.9E-09  1.658797264099121E-02 4.5E-11  1.658797264099121E-02
               11153   5.447403192520142E-01  -4.624237418174744E-01          1.658797450363636E-02          1.658797450363636E-02
               11153  (  1794,     1) (  7958,     1)
          avg abs field values:    1.550020277500153E-02    rms diff: 2.2E-11   avg rel diff(npos):  4.5E-11
                                   1.550020277500153E-02                        avg decimal digits(ndif):  7.1 worst:  6.9
 RMS ua                               2.2055E-11            NORMALIZED  1.4229E-09
```

```
 va   (nele,time)  t_index =      1     1
          1    11153  (     1,     1) (  2066,     1) ( 11113,     1) ( 11113,     1)
               11153   2.148977994918823E+00  -4.877623021602631E-01 3.0E-08  3.256952166557312E-01 8.2E-12  3.256952166557312E-01
               11153   2.148977994918823E+00  -4.877623021602631E-01          3.256952464580536E-01          3.256952464580536E-01
               11153  (     1,     1) (  2066,     1)
          avg abs field values:    2.064534649252892E-02    rms diff: 2.8E-10   avg rel diff(npos):  8.2E-12
                                   2.064534649252892E-02                        avg decimal digits(ndif):  7.0 worst:  7.0
 RMS va                               2.8220E-10            NORMALIZED  1.3669E-08
```

**Comment by uturuncoglu** [comment]
It seems that there are some level of precision loss in here. The pointer that is returned from ESMF fields are r8 (double) but when it is set to internal data structures r4 is used for the temporary variables. To force the FVCOM to use double precision, `-DDOUBLE_PRECISION=ON` is added to compile step of the test. Still testing.

**Comment by uturuncoglu** [comment]
I run the code with `-DDOUBLE_PRECISION=ON` and generate baseline in double data type and checked agains it. The issue is still same. I also set `dbug` option to `true` and wrote incoming wind fields as VTK just after they are imported by FVCOM and they looks identical in all time steps. So, the current hipotestis is having change in those fields after they imported from coupling. This could be in outputting part of the code or setting model internal data structures with the coupled fields. Needs more digging.

**Comment by uturuncoglu** [comment]
@janahaddad @pvelissariou1 @yunfangsun @mansurjisan I started to looking to this case since this is the only regression test that has issue (in terms of reproducibility). As I understand the current RT is not using surface pressure since it is not compiled with `AIR_PRESSURE` cpp option. If you don't mind could you confirm it?

**Comment by uturuncoglu** [comment]
@janahaddad @pvelissariou1 @yunfangsun @mansurjisan Okay I just started to wonder if this is more generic problem in FVCOM. Can anybody could confirm that FVCOM is producing bit-to-bit reproducible answer outside of UFS Coastal? Just run same configuration twice and check the netcdf files in the output directory of both run. I also try to force to use r8 in everywhere with compiler flags but still have difference.

**Comment by mansurjisan** [comment]
Hi @uturuncoglu , I am not very familiar with the FVCOM system but i did run a regression test using the below command:


```./rt.sh -l rt_coastal.conf -a nosofs -k -n "coastal_scituateharbor_atm2fvc intel"```


The code compiled successfully, and I can see the simulation also completed successfully.

Looking at the `stream_data_files01: INPUT/sciww3_forcing_STR_fixed.nc` i can see , there is a pressure variable 

```
ncdump -h INPUT/sciww3_forcing_STR_fixed.nc

        double P(time, latitude, longitude) ;
                P:long_name = "Air Pressure" ;
                P:description = "Sea surface air pressure" ;
                P:units = "Pa" ;
                P:grid = "fvcom_grid" ;
                P:coordinates = "" ;
                P:type = "data" ;
                P:_FillValue = 9.96920996838687e+36 ;
```
Looking at the PET00.ESMF_LogFile i can see pressure variable is getting passed through mediator;

```
PET00.ESMF_LogFile:20250202 104553.244 INFO             PET00 (datm_comp_advertise): Fr_atmSa_pslv
PET00.ESMF_LogFile:20250202 104553.247 INFO             PET00 (med.F90:AdvertiseFields):Fr_atm: Sa_pslv
PET00.ESMF_LogFile:20250202 104553.251 INFO             PET00 (med.F90:AdvertiseFields):To_ocn: Sa_pslv
PET00.ESMF_LogFile:20250202 104554.394 INFO             PET00 (dshr_fldList_realize)(cdeps_datm_comp):(InitializeRealize) :datmExport Field = Sa_pslv is connected using mesh
PET00.ESMF_LogFile:20250202 104554.395 INFO             PET00 (datm_init_dfields): field name = Sa_pslv
PET00.ESMF_LogFile:20250202 104554.446 INFO             PET00 (datm_comp_run) :ES: Sa_pslv  1014.469      1014.511      863328.1         851
PET00.ESMF_LogFile:20250202 104554.446 INFO             PET00 (med_fldList_Realize)(med.F90:RealizeFieldsWithTransferProvided):Fr_atm itemNameList = Sa_pslv
PET00.ESMF_LogFile:20250202 104554.446 INFO             PET00 (med_fldList_Realize)(med.F90:RealizeFieldsWithTransferProvided):Fr_atm Field = Sa_pslv is connected, grid/mesh TBD
PET00.ESMF_LogFile:20250202 104554.447 INFO             PET00 (med_fldList_Realize)(med.F90:RealizeFieldsWithTransferProvided):To_ocn itemNameList = Sa_pslv
PET00.ESMF_LogFile:20250202 104554.447 INFO             PET00 (med_fldList_Realize)(med.F90:RealizeFieldsWithTransferProvided):To_ocn Field = Sa_pslv is connected, grid/mesh TBD
.
.
PET00.ESMF_LogFile:20250202 104554.605 INFO             PET00  Map type bilnr_nstod, destcomp ocn,  mapnorm one  Sa_pslv
PET00.ESMF_LogFile:20250202 104554.612 INFO             PET00 (datm_comp_run) :ES: Sa_pslv  1014.579      1014.624      863419.8         851
PET00.ESMF_LogFile:20250202 104554.617 INFO             PET00 (med_io_write_FB):Sa_pslv:atmImp_Sa_pslv
PET00.ESMF_LogFile:20250202 104554.618 INFO             PET00 (med_io_write_FB):Sa_pslv:ocnExp_Sa_pslv
PET00.ESMF_LogFile:20250202 104609.497 INFO             PET00 (datm_comp_run) :ES: Sa_pslv  1015.120      1015.150      863877.5         851
PET00.ESMF_LogFile:20250202 104609.501 INFO             PET00 (med_io_write_FB):Sa_pslv:atmImp_Sa_pslv
PET00.ESMF_LogFile:20250202 104609.502 INFO             PET00 (med_io_write_FB):Sa_pslv:ocnExp_Sa_pslv
PET00.ESMF_LogFile:20250202 104623.893 INFO             PET00 (datm_comp_run) :ES: Sa_pslv  1015.068      1015.084      863831.4         851
PET00.ESMF_LogFile:20250202 104623.896 INFO             PET00 (med_io_write_FB):Sa_pslv:atmImp_Sa_pslv
PET00.ESMF_LogFile:20250202 104623.898 INFO             PET00 (med_io_write_FB):Sa_pslv:ocnExp_Sa_pslv
PET00.ESMF_LogFile:20250202 104638.095 INFO             PET00 (datm_comp_run) :ES: Sa_pslv  1015.540      1015.582      864239.8         851
```


this is my run directory on Hercules: 

``` /work/noaa/nosofs/mjisan/ufs-weather-model/tests/stmp/mjisan/FV3_RT/rt_35968/coastal_scituateharbor_atm2fvc_intel ```


I can run a standalone version of it. I can see this version of FVCOM https://github.com/FVCOM-GitHub/FVCOM but not sure whether i should use this one or there's another version available.

**Comment by uturuncoglu** [comment]
@mansurjisan Thanks for your help. It would be great to run standalone FVCOM outside of the UFS Coastal two times (same configuration) and compare its output. If you have two run on Hercules just let me know and I could check the output.

**Comment by yunfangsun** [comment]
Hi @uturuncoglu,

Yes, the current `coastal_scituateharbor_atm2fvc` is not using air pressure.

From the namelist `sci_run.nml `of `NML_SURFACE_FORCING`:

```
AIRPRESSURE_ON  = F,
 AIRPRESSURE_KIND        = 'none'                                                                          ,
 AIRPRESSURE_FILE        = 'none'                                                                          ,
 AIRPRESSURE_VALUE       =  0.0000000E+00,
```

**Comment by uturuncoglu** [comment]
@yunfangsun Okay. Thanks for your help.

**Comment by uturuncoglu** [comment]
@yunfangsun BTW, ` WIND_ON = F,` is also set in the nml file. So, do we need to set it true for also coupled application. Maybe those switches are not used by model if it is coupled configuration and just adding flag in the build is sufficient.

**Comment by uturuncoglu** [comment]
@yunfangsun BTW, if I add air pressure to the CPP flag, the case fails as

```
12:  ====== PRINTED NCFILE TYPE ======
12:  ! INITIALIZING SEA SURFACE HEIGHT
12:  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
12:  FVCOM Fatal Error!
12:  COULD NOT FIND VARIABLE 'el_press' IN THE HOTSTART FILE OBJECT
12:  Stopping FVCOM
12:  %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
```

So, I am not able to test the case with pressure at this point. Anyway, we could discuss the issue in the meeting. If we could not solve it, maybe it would be nice to drop FVCOM support at this point.

**Comment by sebinjohn111** [comment]
Hi Ufufk,
I had reported this issue earlier via email. 
If you want to run with the air pressure flag using coastal_scituateharbor_atm2fvc_intel, you may need to create a dummy variable in the restart file (sci_restart_20220621.nc). 

double el_press(time, node) ;
                el_press:long_name = "AIR-Pressure Induced Water Surface Elevation" ;
                el_press:units = "meters" ;

**Comment by yunfangsun** [comment]
Hi @uturuncoglu ,

That is due to that no el_press is available in INPUT/sci_restart_20220621.nc, that required to generate a new restart file.

**Comment by uturuncoglu** [comment]
@sebinjohn111 @yunfangsun Okay. Thanks. Let's wait the results of UFS CAT testing. Then, we could discuss about this issue.

**Comment by uturuncoglu** [comment]
**Adding discussion with Ayumi:**

I am finally back to work on this. I got a couple of hiccups with the libraries but not have a FVCOM executable compiled outside the UFS-Coastal. 
While working on this, I had a couple of questions/notes:

- When FVCOM is compiled within UFS-Coastal, do you know if it uses METIS version 5? In make.inc, I see FLAG411=DMETIS_5 is activated so I assume version 5 is used. However, there's a known issue with makefile about how COBJS is set when FLAG411=DMETIS_5 is set ([link to the discussion](https://github.com/FVCOM-GitHub/FVCOM/issues/17)). This is not fixed in the version within the UFS Coastal so I wondered if it overwrites makefile or make.inc not to use METIS_5.

- In make.inc, I noticed that  -DUSE_NETCDF4 is activated. I think this involves the use of netcdf4 (which is not included in the older version that GLERL is using - v4.3.1). I used Frontera's netcdf module and this did not let me use netcdf4 successfully. So I deactivated -DUSE_NETCDF4. Please let me know if this is going to be an issue. 

- Again in make.inc, I see that FLAG_42=-DPARTITION_SPECIAL is deactivated. For Great Lakes applications, I (and others at CIGLR/GLERL) activate this option. I wonder if this has something to do with the issue of RT not reproducing its results? 

```
#--------------------------------------------------------------------------
#        SPECIAL PARTITION
#        This flag can make sure the identical repeat run for same amount of CPUs 
#--------------------------------------------------------------------------
#             FLAG_42 = -DPARTITION_SPECIAL
```

I will go ahead to run a case twice, and will let you know when it's done. 

Thank you!
Ayumi

**Comment by uturuncoglu** [comment]
**Adding discussion with Ayumi:**

Hi Ayumi,

Thanks for all those feedbacks. UFS Coastal has its own CMake wrapper and that one has no all the options that is supported by the FVCOM but I could add them if there is something missing. Here are my comments,

- We have option in our CMake wrapper like ENABLE_METIS5 and by default it is ON. I did not check it is used by not since I have no information related with the bug. Let me know if we could fix it in our end. 

https://github.com/oceanmodeling/ufs-weather-model/blob/0eb40dd5024ab2903ae188490e21abfb6d466f8b/FVCOM-interface/CMakeLists.txt#L56

- We were assuming to use USE_NETCDF4 when netcdf found but maybe I could make it optional. We could check netcdf4 support explicitly before defining this option. This is todo in my end. At this point, deactivating it manually will not harm.

- This option is missing in our CMake wrapper. As I understand from the information, it is important for reproducibility. Let me set that one an try to test. I’ll update you if it solves the issue.

Again, thanks for all your help.

Best,

—ufuk

**Comment by uturuncoglu** [comment]
**Note from extra test:** 

Even with -DPARTITION_SPECIAL option the model does not produce itself under UFS Coastal. Let me know if you have some result from your tests.

**Comment by ayumifuji** [comment]
@uturuncoglu 
I completed the test. The results are below on Frontera.

/work2/09028/ayumif/frontera/ufs-coastal/fvcom-standalone/run/OUTPUT_6936085
/work2/09028/ayumif/frontera/ufs-coastal/fvcom-standalone/run/OUTPUT_6936196

They are the results obtained with the same FVCOM executable (/work2/09028/ayumif/frontera/ufs-coastal/fvcom-standalone/FVCOM-interface/FVCOM/src/fvcom), same input files taken from  /work2/01118/tg803972/frontera/RT/NEMSfv3gfs/input-data-20240501/FVCOM/Sci/atm2fvc/, and same # of cores (30). 

I also has tmp.nc, which is a result of "ncdiff OUTPUT_6936196/sci_0001.nc OUTPUT_6936085/sci_0001.nc tmp.nc". I quickly checked "temp" in tmp.nc (the difference between the two output files), and they look all zero. I think the model was run with -DDOUBLE_PRECISION -DSINGLE_OUTPUT so there might be difference at the digits that are not captured by single precision, which is used for outputs. 
 
Please let me know if you have questions and/or like me to run additional tests!
Ayumi

**Comment by uturuncoglu** [comment]
@ayumifuji Thanks for your help. Let me quickly check the results with the tools. I also wonder if you could share your build log. So, if we are missing some option in the build that needs to be there, we could catch with this way.

**Comment by uturuncoglu** [comment]
@ayumifuji I am getting permission error when I try to access the folders. I could see the content of the `/work2/09028/ayumif/frontera/ufs-coastal/` but not after that.

**Comment by ayumifuji** [comment]
@uturuncoglu Sorry! I forgot to update the permission. Now it should be viewable & executable (755). Please let me know if it still does not work. I will also create a build log and will let you know.

**Comment by ayumifuji** [comment]
Here's the build log: /work2/09028/ayumif/frontera/ufs-coastal/fvcom-standalone/FVCOM-interface/FVCOM/src/build_fvcom.log

**Comment by uturuncoglu** [comment]
@ayumifuji Thanks. I could access to the files now without any issue. Thanks for the log file too. I'll update you when I check them.

**Comment by uturuncoglu** [comment]
@ayumifuji Again thanks for your help and also sorry for late response. I have just checked two runs with `cprnc` (special tool developed by NCAR) and it seems that results of both runs are different. Here is the part of the output for your reference,

```
 file 1=OUTPUT_6936085/sci_0001.nc
 file 2=OUTPUT_6936196/sci_0001.nc
...
...
SUMMARY of cprnc:
 A total number of   1778 fields were compared
          of which    663 had non-zero differences
               and      0 had differences in fill patterns
               and      0 had different dimension sizes
               and      0 had different data types
 A total number of    146 fields could not be analyzed
 A total number of      0 time-varying fields on file 1 were not found on file 2.
 A total number of      0 time-constant fields on file 1 were not found on file 2.
 A total number of      0 time-varying fields on file 2 were not found on file 1.
 A total number of      0 time-constant fields on file 2 were not found on file 1.
  diff_test: the two files seem to be DIFFERENT 
```

It seems that the first time step is fine and identical but after that the values start to diverge. As I understand these runs are outside of UFS Coastal and they are not using CDEPS for forcing. So, all standalone FVCOM with its own build. Right? If so, then this confirm my previous comments and something in the FVCOM model is not right and preventing to get bit-to-bit identical results. Anyway, if it is easy, can you do the testing again with most recent version of the FVCOM to see what happens. If that one is not identical too then I think we can't do anything for FVCOM in our end and its developers need to fix this issue. Maybe we are missing some CPP switch etc. but someone from FVCOM side needs to convince me and show me two identical run with same answers at this point.

@janahaddad @saeed-moghimi-noaa If the test with recent version shows also same issue then I am planing to deactivate FVCOM from regular RT testing for now until it is fixed by the FVCOM developers. This will not prevent anyone to activate that test and run the RT to create another configuration under UFS Coastal. It just won't make sense to run RT for FVCOM at this point and this will also help in our Jenkins workflow since FVCOM was always failing in all the platforms. Please let me know if you are thinking similar with me and have any suggestion.

**Comment by ayumifuji** [comment]
@uturuncoglu 
Thank you for the update. Yes, you are correct that these runs are from all standaline FVCOM, outside of UFS Coastal, not using CDEPS for forcing.
I noticed that I forgot to note that I was not using  -DPARTITION_SPECIAL. I will compile it again and rerun it now. Sorry for the oversight. Not sure if this helps (since you tested it via UFS-Coastal, I think?), but it's probably good to check. 

Regarding testing the latest version of FVCOM, I will ask UFS-Coastal FVCOM testers if they could help. I could potentially help but they are probably faster. 

Best,
Ayumi

**Comment by uturuncoglu** [comment]
@ayumifuji Thanks for your help. Yes, I agree. Checking with `-DPARTITION_SPECIAL` would be great. If it works in your end then I could also double check under UFS Coastal. Let me know when you have two run with new option and I could check the results. Best

**Comment by ayumifuji** [comment]
@uturuncoglu 
Here are the new output directories with -DPARTITION_SPECIAL: 
/work2/09028/ayumif/frontera/ufs-coastal/fvcom-standalone/run/OUTPUT_6963092
/work2/09028/ayumif/frontera/ufs-coastal/fvcom-standalone/run/OUTPUT_6962974

**Comment by uturuncoglu** [comment]
@ayumifuji It seems that `-DPARTITION_SPECIAL` is the option that we are looking for. The results are matching in this case. If I remember it was not working under UFS Coastal but let me try again with fresh clone to double check. I hope it will work in there too. Again, thanks for your help and support. I'll update you soon.

**Comment by uturuncoglu** [comment]
@ayumifuji If you don't mind could you also point your build log. I am getting premmision issue from `/work2/09028/ayumif/frontera/ufs-coastal/fvcom-standalone/FVCOM-interface/FVCOM/src/build_fvcom.log` but maybe you have newer one.

**Comment by ayumifuji** [comment]
@uturuncoglu not sure why permission started not working. The build_fvcom.log was from my old compile and was not changed. Anyway, I created the log in the file below. The permission should be good but please let me know if you have access issues still. 

/work2/09028/ayumif/frontera/ufs-coastal/fvcom-standalone/FVCOM-interface/FVCOM/src/build_fvcom_DPARTITION_SPECIAL.log

**Comment by uturuncoglu** [comment]
@ayumifuji @janahaddad @saeed-moghimi-noaa @gseroka I double checked the FVCOM RT under UFS Coastal with `-DPARTITION_SPECIAL` CPP option and it seems that it is solving the reproducibility issue but I need to do also following to have a clean RT run.

- There is a variable called `file_date` in the FVCOM model output (both history and restart). Since it is changing all the time with a new run, It leads to a failure in the baseline check step. So, I removed that variable from the history files by setting `NC_FILE_DATE` to `F` in the `NML_NETCDF` section of the `sci_run.nml` configuration file. Unfortunately, this option is not available for the restart file. So, I modified FVCOM code (in our fork) to force not to add `file_date` variable also to restart file when `NC_FILE_DATE` is set to `F`. This is just a minor fix in our end and it is working fine.

- I also add `CMP_DATAONLY=true` option to `tests/coastal_scituateharbor_atm2fvc` RT definition file to prevent checking netcdf global attributes (this is also used in ROMS tests) since it has time stamp metadata that changes every run.

- I added `-DPARTITION_SPECIAL` to FVCOM CMake interface in UFS Coastal level. 
 
- I updated FVCOM baseline on Hercules (will do on Frontera soon, I also need to add `NC_FILE_DATE` to `F` in the `NML_NETCDF` section in there too). So, FVCOM RT is clean on Hercules now. It is a great news!!!

I also want to thank to @ayumifuji again for her great help and support to find the underlying issue. It was impossible for me to find it without her help. So, I am closing this long standing ticket with help from her. Thanks again.

**Comment by janahaddad** [comment]
Thanks so much @ayumifuji and @uturuncoglu
